algebird-test/src/main/scala/com/twitter/algebird/ApproximateProperty.scala

```diff
@@ -0,0 +1,96 @@
+package com.twitter.algebird
+
+import org.scalacheck.{ Gen, Prop }
+import org.scalacheck.util.Pretty
+
+trait ApproximateProperty {
+  type Exact
+  type Approx
+  type Input
+  type Result
+
+  def exactGenerator: Gen[Exact]
+  def inputGenerator(e: Exact): Gen[Input]
```

Comment:
Exact is useful here so that we can generate appropriate test inputs. 

For example, suppose we're testing CMS[Long]. We need to generate some inputs so that we can query `exactList.frequency(x)` and `cms.frequency(x)` and compare them. 
We could simply query using random Long values. However, most of the random Long values we generate would probably not be in the list, so these test cases would not be very useful. If we know the exact value, we can choose random elements in `exactList` as our test cases.

I'm not sure if this is the best way to do this.